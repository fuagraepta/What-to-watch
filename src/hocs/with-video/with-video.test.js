import React from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';

import withVideo from './with-video.js';

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
};

describe(`The component with video is rendered correctly`, () => {
  it(`With video correctly renders`, () => {
    const MockComponentWrapped = withVideo(MockComponent);

    const mockComponentWrapped = renderer
    .create(
        <MockComponentWrapped
          isPlaying = {false}
          style = {{width: `100%`, height: `100%`, objectFit: `fill`}}
        />
    )
    .toJSON();

    expect(mockComponentWrapped).toMatchSnapshot();
  });
});
