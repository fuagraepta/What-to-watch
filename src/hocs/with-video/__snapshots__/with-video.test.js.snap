// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component with video is rendered correctly With video correctly renders 1`] = `<div />`;
